src/Microsoft.AspNetCore.Http/Internal/BindingAddress.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Http.Internal
+{
+    public class BindingAddress
+    {
+        public string Host { get; private set; }
+        public string PathBase { get; private set; }
+        public int Port { get; internal set; }
+        public string Scheme { get; private set; }
+
+        public bool IsUnixPipe
+        {
+            get
+            {
+                return Host.StartsWith(Constants.UnixPipeHostPrefix, StringComparison.Ordinal);
+            }
+        }
+
+        public string UnixPipePath
+        {
+            get
+            {
+                Debug.Assert(IsUnixPipe);
```

Comment:
Now that this is in another assembly it might be better to just change this to an if/throw.